src/osd/OSD.cc

```diff
@@ -7292,25 +7292,20 @@ void OSDService::maybe_share_map(
 	     << " -> " << peer_epoch_lb << " (as per caller)" << dendl;
     session->last_sent_epoch = peer_epoch_lb;
   }
-  epoch_t last_sent_epoch = session->last_sent_epoch;
```

Comment:
The commit message in this commit needs to be explicit about why this is allowed.  Invoking send_incremental_map doesn't wait until the replica actually receives the map, so marking last_epoch_sent prior to invoking it is just as correct.